lib/src/sledds/datastore.rs

```diff
@@ -0,0 +1,494 @@
+use super::super::{
+    Datastore, EdgeDirection, EdgePropertyQuery, EdgeQuery, Transaction, VertexPropertyQuery, VertexQuery,
+};
+use super::managers::*;
+use crate::errors::Result;
+use crate::models;
+use crate::models::*;
+use crate::util::next_uuid;
+use chrono::offset::Utc;
+use serde_json::Value as JsonValue;
+use sled::{Db, Tree};
+use std::sync::Arc;
+use std::u64;
+use std::usize;
+use uuid::Uuid;
+
+fn remove_nones_from_iterator<I, T>(iter: I) -> impl Iterator<Item = Result<T>>
+where
+    I: Iterator<Item = Result<Option<T>>>,
+{
+    iter.filter_map(|item| match item {
+        Err(err) => Some(Err(err)),
+        Ok(Some(value)) => Some(Ok(value)),
+        _ => None,
+    })
+}
```

Comment:
Could you move this method to `lib/src/util.rs`, and reuse it both here and in the rdb implementation?

Reply:
Yep, I can do that